lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -77,6 +88,14 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
+      %w[average maximum minimum].each do |m_name|
+        define_method("define_virtual_#{m_name}_method") do |name, relation, column|
+          define_method(name) do
+            (has_attribute?(name) ? self[name] : send(relation).try(m_name.to_sym, column)) || 0
```

Comment:
I think we may want to handle the case where the relation is loaded but the field is not. see for the size method.

It is too bad the size method takes no parameters, otherwise we could collapse all 4

Reply:
not marking as resolved b/c i haven't addressed this yet

Reply:
ok, if these are all the same, then no reason for duplicating the private method here.

Just call the contents of this directly from virtual_average or something